tests/test_data.py

```diff
@@ -65,3 +73,76 @@ def test_can_instantiate_filedataset(test_filedata, test_csv):
 def test_cannot_instantiate_an_abstract_baseclass():
     with pytest.raises(TypeError):
         Dataset()
+
+
+def test_dataset_can_copy_to_table(
+    test_sqldata, test_sqldata_class, test_db, test_table
+):
+    copy_to_data = test_sqldata_class(test_db, "main")
+    copy_to_data.table = test_table
+
+    test_sqldata.table = test_table
+    test_sqldata.copy_to(target=copy_to_data)
+
+    assert copy_to_data._load_prediction_data().index.all(
+        [
+            "CRIM",
+            "ZN",
+            "INDUS",
+            "CHAS",
+            "NOX",
+            "RM",
+            "AGE",
+            "DIS",
+            "RAD",
+            "TAX",
+            "PTRATIO",
+            "B",
+            "LSTAT",
+        ]
+    )
+
+
+@patch("ml_tooling.data.sql.SQLDataset.create_connection")
+def test_dump_data_throws_error_on_exec_failure(
+    create_conn_mock, test_sqldata, test_table
+):
+    test_sqldata.table = test_table
+
+    failing_conn_mock = MagicMock()
+    failing_conn_mock.execute.side_effect = DBAPIError("test", "test", "test")
+    create_conn_mock.return_value.__enter__.return_value = failing_conn_mock
+    with pytest.raises(DBAPIError):
+        test_sqldata._dump_data()
+
+
+@patch("ml_tooling.data.sql.SQLDataset.create_connection")
+def test_insert_data_throws_error_on_exec_failure(
+    create_conn_mock, test_sqldata, test_table
+):
+    test_sqldata.table = test_table
+
+    failing_conn_mock = MagicMock()
+    failing_conn_mock.execute.side_effect = DBAPIError("test", "test", "test")
+    trans_mock = MagicMock()
+
+    create_conn_mock.return_value.__enter__.return_value = failing_conn_mock
+    create_conn_mock.begin.return_value = trans_mock
+
+    with pytest.raises(DBAPIError):
+        test_sqldata._insert_data(MagicMock())
+        trans_mock.rollback.assert_called()
```

Comment:
```suggestion
    trans_mock.rollback.assert_called()
```

The `raises` block should only contain the code that raises, to avoid false positives